### PR TITLE
Updates to Plugin for Android

### PR DESCRIPTION
**Pull request by bhariharan**
Next pull request will have the additions for moving ```Gradle``` files and libraries around to make it work with ```Android Studio```.

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
   <dependency id="cordova-plugin-whitelist" />
   <dependency id="cordova-plugin-device" />
-  <dependency id="com.phonegap.plugins.PushPlugin" url="https://github.com/forcedotcom/PushPlugin" />
+  <dependency id="cordova-plugin-push" url="https://github.com/forcedotcom/phonegap-plugin-push" />
```
Comment (bhariharan) [inline plugin.xml]: New push plugin. Remains to be seen if we need custom hooks in our fork.

File: tools/preinstall-android.js [hunk 2]
```diff
@@ -1,12 +0,0 @@
-if (process.version < 'v0.12')
```
Comment (bhariharan) [inline tools/preinstall-android.js]: No more pre-install script, since we don't use ```ant``` anymore.

File: tools/postinstall-android.js [hunk 3]
```diff
@@ -1,5 +1,4 @@
 console.log("Running SalesforceMobileSDK plugin android post-install script");
-var useSmartStoreOrSmartSync = true;
```
Comment (bhariharan) [inline tools/postinstall-android.js]: Removed this, since it's always ```true``` (has been for a while now).

File: tools/postinstall-android.js [hunk 4]
```diff
@@ -87,40 +77,10 @@ if (androidExePath === null) {
...
 fixFile(path.join('platforms', 'android', 'AndroidManifest.xml'), fixAndroidManifest);
 
-console.log('Fixing application project.properties');
-fixFile(path.join('platforms', 'android', 'project.properties'), fixProjectProperties);
-
-console.log('Removing cordova library project reference from SalesforceSDK\'s project.properties');
-fixFile(path.join('plugins', 'com.salesforce', 'src', 'android', 'libs', 'SalesforceSDK', 'project.properties'), fixSDKProjectProperties);
-
-console.log('Building cordova library');
-exec('ant debug', {cwd: path.resolve(process.cwd(), path.join('platforms', 'android', 'CordovaLib'))});
```
Comment (bhariharan) [inline tools/postinstall-android.js]: None of these files exist anymore. Will be replaced with ```Gradle``` updates.

**Comment by bhariharan** [comment]
Any feedback? Can I merge this in?

**Comment by kchitalia** [comment]
Looks good to me :+1:

**Pull request merged**